### DISCUSSION TITLE
The autocomplete popup should not appear on typing a left square bracket if no specific suggestions are available

### DISCUSSION THREAD

**Discussion by atrigent** [microsoft/pylance-release#3236] [Enhancement]
Yes, I know this is similar to the pre-existing issue #1712 and the discussion #2599 that was created from it. However, I'm not asking for the popup to only contain relevant suggestions or those that would be syntactically valid - I'm suggesting that the popup should not appear at all in this context. I believe it does because, in some cases when typing a `[`, such as when indexing a dictionary that was defined nearby or for which type information is available, Pylance is able to make specific, useful suggestions. However, in most other situations it can't. Pylance doesn't open the autocomplete popup in any other situations where no particular information or specific suggestions are available, because it's more reasonable then to wait until the user starts typing a name or explicitly invokes the autocomplete dialog. Pylance should do the same in such cases when `[` is typed.

<h3> Environment data </h3>

-   Language Server version: 2022.8.30
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.8
-   python.analysis.indexing: null
-   python.analysis.typeCheckingMode: off

<h3> Repro Steps </h3>

1. Type a `[` in an editor where pylance is activated in a situation where no specific suggestions are available
2. See a not particularly useful autocomplete popup appear

<h3> Expected behavior </h3>

This popup should not appear, just as it doesn't in any other analogous situations.

<h3> Actual behavior </h3>

A popup appears with some generic suggestions.

**Comment by debonte** [comment]
I'm guessing that the user is seeing the keyword and stdlib suggestions, like this:
![image](https://user-images.githubusercontent.com/12820898/185973435-d5478863-082a-4ab8-a96b-db369c57617d.png)

@atrigent, can you confirm? If you're seeing something different, please provide a code sample and a screenshot of the completion list you are getting.

**Comment by atrigent** [comment]
Yes, that is what I'm seeing. It's the same thing the person was seeing in #1712.

**Comment by judej** [comment]
This would be an enhancement and will therefore move to discussions for more user feedback/upvotes.

**Comment by atrigent** [comment]
@judej I disagree that this is an enhancement or that it requires a discussion post. Did you read my description of the issue? Even if this behavior was intentional (and I don't believe it was), it's inconsistent with the rest of the behavior of the plugin and should be changed for that reason. It also simply makes no sense to start autocompleting before there is any indication at all what the user intends to type.

Please change this back to an issue.

**Comment by SeanWelshBrown** [comment]
I agree with atringent on this one-- this is a strange behavior that seems at odds with the way the way the rest of the plugin works, and most certainly should not be occurring. I've been trying to find some way to fix the issue, only to see that anything relating to it has been moved here to the "discussion" section, which doesn't seem right.

**Comment by rchiodo** [comment]
I made an issue out of this discussion. I think it looks weird too, but maybe there's a technical reason why we can't have an empty list. We'll discuss in triage next week.

**Reply by SeanWelshBrown** [reply]
Thanks for the fast action/response on this! It's a small detail, but it's the sort of thing that can really stick out when the rest of the functionality feels quite smooth and intentional. Would be very interested to hear if there's some kind of technical reason behind it. 👍
